ext/bg/js/search-query-parser.js

```diff
@@ -99,11 +102,23 @@ class QueryParser extends TextScanner {
         this.queryParser.dataset.termSpacing = `${options.parsing.termSpacing}`;
     }
 
+    getOptionsContext() {
```

Comment:
These declarations are redundant since the properties are assigned in the constructor. Adding a doc comment (#403) which describes the expected arguments for the functions could be helpful. If you want to make sure that the programmer has assigned proper values, better to verify that in the constructor rather than later on. I.e.
```js
if (typeof getOptionsContext !== 'function') { throw new Error('Invalid value for getOptionsContext'); }
```

Reply:
You're right, I think I'll remove the stubs and don't add type checking for now because the class is only used in one place.